src/na/NAGraphAlgorithms.cpp

```diff
@@ -188,11 +188,42 @@ auto NAGraphAlgorithms::colorEdges(
               [&](const Edge& a, const Edge& b) {
                 const auto u = a.first == v ? a.second : a.first;
                 const auto w = b.first == v ? b.second : b.first;
-                return partialOrder.isReachable(u, w) or
-                       (!partialOrder.isReachable(w, u) and
-                        (nAdjColors[a] > nAdjColors[b] or
-                         (nAdjColors[a] == nAdjColors[b] and
-                          edgeDegree[a] > edgeDegree[b])));
+                if (u == w) {
+                  // the compare function defines a proper less than relation,
+                  // i.e., equal elements must return false
+                  return false;
```

Comment:
This is not yet covered by tests as well. Would that be hard to do?

Reply:
So, this is only written this way with all the if-else branches because clang-tidy was warning because of a long condition. I tried to take benefit of the new structure and added helpful comments for each case. Additionally, testing it, is hard because I would need to engeneer for each case a special circuit and then it also depends on the randomness of memory addresses. I am very hesitating to do that now, in particular as the logica of the code did not change really.

Reply:
I really like the new structure of this.
Makes things quite explicit 👍🏻

And just to make that clear: it is only this single if statement that is not yet covered. All the others are covered already according to the data.

However, I fully understand that you do not want to spend a substantial amount of time to engineer an example.
If not quickly possible, then feel free to just resolve this conversation 😌